notebooks/Solution_Exercises.ipynb

```diff
@@ -0,0 +1,165 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "47d19053",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Question 1: Read the geojson file:\n",
+    "import json\n",
+    "with open(geojson_file) as f:\n",
+    "    d = json.load(f)\n",
+    "    s = json.dumps(d)\n",
+    "values = geojson.loads(s)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e8fa7860",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Question 2: Overlay the shapes from a specific plane on the image\n",
+    "z = 5\n",
+    "fiz, ax = plt.subplots()\n",
+    "plt.imshow(data[0, 0, z, :, :], cmap='gray')\n",
+    "for v in values:\n",
+    "    if v.properties[\"z\"] == z:\n",
+    "        label_geojson = numpy.array(v.geometry.coordinates)\n",
+    "        p = label_geojson.squeeze()\n",
+    "        ax.plot(p[:, 0], p[:, 1], linewidth=v.properties[\"stroke-width\"], color=\"red\")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dddaad7f",
+   "metadata": {},
+   "source": [
+    "### Exercise 2 \n",
+    "\n",
+    "In order to save the StarDist labels to an OMERO.server with write access, you will first need to log in and change the image ID."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "e0d0c98b",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "False\n"
+     ]
+    }
+   ],
+   "source": [
+    "from omero.gateway import BlitzGateway\n",
+    "HOST = 'YOUR_SERVER'\n",
+    "conn = BlitzGateway('YOUR_USERNAME', 'YOUR_PASSWORD',\n",
+    "                    host=HOST, secure=True)\n",
+    "print(conn.connect())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3f3f0360",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Question 1: Convert the StarDist labels into OMERO polygons\n",
+    "import omero\n",
+    "from omero.rtypes import rint, rstring\n",
+    "\n",
+    "image = omero.model.ImageI(image_id, False)\n",
+    "for i in range(len(img)):\n",
+    "    new_labels, details = model_versatile.predict_instances(img[i])\n",
+    "    # Convert into Polygon\n",
+    "    rois = []\n",
+    "    for obj_id, region in enumerate(details['coord']):\n",
+    "        roi = omero.model.RoiI()\n",
+    "        roi.setName(rstring(\"Object %s\" % obj_id))\n",
+    "        polygon = omero.model.PolygonI()\n",
+    "        x = region[1]\n",
+    "        y = region[0]\n",
+    "        points = \" \".join(\n",
+    "            [\"{},{}\".format(x[j], y[j]) for j in range(len(x))])\n",
+    "        polygon.setPoints(rstring(points))\n",
+    "        polygon.theZ = rint(i)\n",
+    "        polygon.theC = rint(0)\n",
+    "        roi.addShape(polygon)\n",
+    "        # Link the ROI and the image\n",
+    "        roi.setImage(image)\n",
+    "        rois.append(roi)                    "
```

Comment:
this principally works and the rois are saved. But please check https://outreach.openmicroscopy.org/webclient/?show=image-55211 - all the ROIs (only 7, which seems too low a number) are associated with z plane 257, which is the last one.

Reply:
Fixed now